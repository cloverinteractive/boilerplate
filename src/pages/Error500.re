[@react.component]
let make = () =>
  <div className="container is-fluid">
    <h1 className="title"> "Something went wrong"->React.string </h1>
    <div className="content">
      <p>
        "Something went wrong with the operation you tried, please try again later"
        ->React.string
      </p>
    </div>
    <BsReactHelmet title="Something went wrong" />
  </div>;

let default = make;