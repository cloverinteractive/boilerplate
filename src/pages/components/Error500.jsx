import * as React from 'react';
import { Helmet } from 'react-helmet';

const Error500 = () => (
  <div className="container is-fluid">
    <h1 className="title">Something</h1>
    <div className="content">
      <p>Something went wrong with the operation you tried, please try again later</p>
    </div>
    <Helmet>
      <title>Something went wrong</title>
    </Helmet>
  </div>
);

export default Error500;
