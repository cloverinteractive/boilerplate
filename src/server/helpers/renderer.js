import { renderToString } from 'react-dom/server';
import { Helmet } from 'react-helmet';
import serialize from 'serialize-javascript';
import { isDevelopment } from 'server/helpers/env-access';

export default (Component, store) => {
  const content = renderToString(Component);
  const helmet = Helmet.renderStatic();

  const productionStyles =
    `<link rel="stylesheet" href="/vendors.css">
    <link rel="stylesheet" href="/styles.css">`;

  const styles = isDevelopment ? '' : productionStyles;

  const template =
    `<!doctype html>
    <html lang="en-US">
      <head>
        <meta charset="utf-8">
        ${helmet.link.toString()}
        ${helmet.title.toString()}
        ${helmet.meta.toString()}
        ${styles}
        <script src="/vendors.js"></script>
      </head>
      <body>
        <div id="app">${content}</div>
        <script>
          window.__INITIAL_STATE__ = ${serialize(store.getState())};
        </script>
        <script async src="/bundle.js"></script>
      </body>
    </html>`;

  return template;
};
