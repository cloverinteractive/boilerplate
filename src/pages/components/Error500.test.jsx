import * as React from 'react';
import renderer from 'react-test-renderer';
import Error500 from './Error500';

describe('<Error500 />', () => {
  test('it renders correctly', () => {
    const tree = renderer
      .create(<Error500 />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
