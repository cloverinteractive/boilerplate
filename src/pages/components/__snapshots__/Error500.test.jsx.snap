// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Error500 /> it renders correctly 1`] = `
<div
  className="container is-fluid"
>
  <h1
    className="title"
  >
    Something
  </h1>
  <div
    className="content"
  >
    <p>
      Something went wrong with the operation you tried, please try again later
    </p>
  </div>
</div>
`;
